File: test_data/code/java/1556-thousand-separator.java
class Solution {
    public String thousandSeparator(int n) {
        if (n < 1000) {
            return String.valueOf(n);
        }
        String str = String.valueOf(n);
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < str.length(); i++) {
            sb.append(str.charAt(i));
            if ((i + 1) % 3 == 0 && i != str.length() - 1) {
                sb.append('.');
            }
        }
        return sb.toString();
    }
}

File: test_data/code/python/1556-thousand-separator.py
class Solution:
    def thousandSeparator(self, n: int) -> str:
        n_str = str(n)
        result = ""
        for i, char in enumerate(n_str[::-1]):
            if i % 3 == 0 and i != 0:
                result = "." + result
            result = char + result
        return "0" + result if n != 0 else "0"